### DISCUSSION TITLE
Interface "Psr\Http\Message\RequestInterface" not found

### DISCUSSION THREAD

**Discussion by ZoeFken** [mpdf/mpdf#1839] [Q&A]
I am by no means a experienced developer and or programmer. 

I was revamping a old project of mine to work with php 8.* when I got alot of errors from mpdf. Me thinking I could solve most of my issues by upgrading to the latest version was wrong.

This error I am getting now is flabbergasting me.

An uncaught Exception was encountered
Type: Error

Message: Interface "Psr\Http\Message\RequestInterface" not found

Filename: C:\xampp\htdocs\vbuwdoc\vendor\mpdf\mpdf\src\Http\Request.php

Line Number: 13

Anyone who can point me in the right direction?

**Comment by finwe** [comment]
You need to install dependencies of mPDF using composer. 

https://getcomposer.org/doc/01-basic-usage.md
https://github.com/mpdf/mpdf#installation

**Reply by ZoeFken** [reply]
```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - setasign/fpdi is locked to version v2.3.1 and an update of this package was not requested.
    - setasign/fpdi v2.3.1 requires php ^5.6 || ^7.0 -> your php version (8.2.0) does not satisfy that requirement.
  Problem 2
    - phpunit/phpunit is locked to version 5.7.27 and an update of this package was not requested.
    - phpunit/phpunit 5.7.27 requires php ^5.6 || ^7.0 -> your php version (8.2.0) does not satisfy that requirement.

You can also try re-running composer require with an explicit version constraint, e.g. "composer require mpdf/mpdf:*" to figure out if any version is installable, or "composer require mpdf/mpdf:^2.1" if you know which you need.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

	"require": {
		"php": ">=5.3.7",
		"setasign/fpdf": "1.8.*",
		"setasign/fpdi": "^2.0",
		"mpdf/mpdf": "^8.0",
		"smalot/pdfparser": "*"
	},
	"suggest": {
		"paragonie/random_compat": "Provides better randomness in PHP 5.x"
	},
	"require-dev": {
		"mikey179/vfsstream": "1.1.*",
		"phpunit/phpunit": "4.* || 5.*"
	}
```

**Reply by finwe** [reply]
on what action? what is the source of the composer.json at the end of the comment?
